http3/client.go

```diff
@@ -251,6 +251,10 @@ func (c *client) RoundTripOpt(req *http.Request, opt RoundTripOpt) (*http.Respon
 		return nil, c.handshakeErr
 	}
 
+	req.Proto = "HTTP/3.0"
+	req.ProtoMajor = 3
+	req.ProtoMinor = 0
```

Comment:
Why do we need to set these? The documentation for `http.Request` says:
> For client requests, these fields are ignored.

Reply:
> If we return it, it should be set correctly though

Otherwise, the request added to the response will say it's http1

Reply:
I was just wondering what happens if you use the same `http.Request` struct multiple times, potentially on HTTP/3 and HTTP/2. Are we creating a race condition here by modifying the struct?